### PR TITLE
chore:use serializer instead of codec

### PR DESCRIPTION
**Pull request by leizhiyuan**
### Ⅰ. Describe what this PR did

codec is not a good name for this scene.

### CODE REVIEW

File: core/src/main/java/io/seata/core/rpc/netty/v1/ProtocolV1Encoder.java [hunk 1]
```diff
@@ -91,8 +91,8 @@ public void encode(ChannelHandlerContext ctx, Object msg, ByteBuf out) {
                 if (messageType != ProtocolConstants.MSGTYPE_HEARTBEAT_REQUEST
                         && messageType != ProtocolConstants.MSGTYPE_HEARTBEAT_RESPONSE) {
                     // heartbeat has no body
-                    Codec codec = CodecFactory.getCodec(rpcMessage.getCodec());
-                    bodyBytes = codec.encode(rpcMessage.getBody());
+                    Serializer codec = SerializerFactory.getSerializer(rpcMessage.getCodec());
```
Comment (l81893521) [inline core/src/main/java/io/seata/core/rpc/netty/v1/ProtocolV1Encoder.java]: should change the variable name to serializer

File: core/src/main/java/io/seata/core/rpc/netty/v1/ProtocolV1Decoder.java [hunk 2]
```diff
@@ -137,8 +137,8 @@ public Object decodeFrame(ByteBuf frame) {
                 frame.readBytes(bs);
                 Compressor compressor = CompressorFactory.getCompressor(compressorType);
                 bs = compressor.decompress(bs);
-                Codec codec = CodecFactory.getCodec(codecType);
-                rpcMessage.setBody(codec.decode(bs));
+                Serializer codec = SerializerFactory.getSerializer(codecType);
```
Comment (l81893521) [inline core/src/main/java/io/seata/core/rpc/netty/v1/ProtocolV1Decoder.java]: should change the variable name to serializer.

File: codecov.yml [hunk 3]
```diff
@@ -11,7 +11,7 @@ coverage:
   precision: 2
   range: "50...100"
 ignore:
- - "codec/seata-codec-protobuf/src/main/java/io/seata/codec/protobuf/generated"
+ - "codec/seata-serializer-protobuf/src/main/java/io/seata/codec/protobuf/generated"
```
Comment (l81893521) [inline codecov.yml]: Is it the path of codec should change to serializer too?

Comment (leizhiyuan) [reply codecov.yml]: > Is it the path of codec should change to serializer too?

ok

File: serializer/seata-serializer-hessian/src/test/java/io/seata/serializer/hessian/HessianSerializerTest.java [hunk 4]
```diff
@@ -29,13 +29,13 @@
 /**
  * @author Xin Wang
  */
-public class HessianCodecTest {
+public class HessianSerializerTest {
 
-    private static HessianCodec hessianCodec;
+    private static HessianSerializer hessianCodec;
```
Comment (slievrly) [inline serializer/seata-serializer-hessian/src/test/java/io/seata/serializer/hessian/HessianSerializerTest.java]: hessianCodec -> serializer

File: serializer/seata-serializer-kryo/src/test/java/io/seata/serializer/kryo/KryoSerializerTest.java [hunk 5]
```diff
@@ -29,13 +29,13 @@
 /**
  * @author jsbxyyx
  */
-public class KryoCodecTest {
+public class KryoSerializerTest {
 
-    private static KryoCodec kryoCodec;
+    private static KryoSerializer kryoCodec;
```
Comment (slievrly) [inline serializer/seata-serializer-kryo/src/test/java/io/seata/serializer/kryo/KryoSerializerTest.java]: kryoCodec -> serializer

**Pull request merged**